src/sentinel.c

```diff
@@ -3450,12 +3460,25 @@ void addReplySentinelRedisInstance(client *c, sentinelRedisInstance *ri) {
 void addReplyDictOfRedisInstances(client *c, dict *instances) {
     dictIterator *di;
     dictEntry *de;
+    int ds;
+
+    /* count elements excluding unannonced replicas */
+    ds = 0;
+    di = dictGetIterator(instances);
+    while((de = dictNext(di)) != NULL) {
+        sentinelRedisInstance *ri = dictGetVal(de);
+        if (ri->flags & SRI_SLAVE && ri->replica_announced == 0) continue;
+        ds++;
+    }
+    dictReleaseIterator(di);
+    addReplyArrayLen(c,ds);
```

Comment:
actually you don't need to loop through the instance dict twice, you can use addReplyDeferredLen and set the length later, this will skip one unnecessary while loops

Reply:
fixed, thanks for the tip